keras/layers/core.py

```diff
@@ -163,10 +163,12 @@ def get_config(self):
 class Merge(Layer):
     def __init__(self, layers, mode='sum'):
         ''' Merge the output of a list of layers or containers into a single tensor.
-            mode: {'sum', 'concat'}
+            mode: {'sum', 'mul', 'concat'}
         '''
         if len(layers) < 2:
             raise Exception("Please specify two or more input layers (or containers) to merge")
+        elif len(layers) > 2 and mode == 'mul':
+            raise Exception("Elemwise multiplication mode can only merge two layers")
```

Comment:
This should definitely be changed to support N inputs (at no loss of functionality...), much like `sum`.